aten/src/ATen/native/cuda/Sorting.cu

```diff
@@ -316,6 +316,9 @@ std::tuple<Tensor&, Tensor&> median_with_indices_impl(
     int64_t dim,
     bool keepdim,
     bool ignore_nan) {
+  // See note [Writing Nondeterministic Operations]
+  // Nondeterministic if indices output contains duplicates
```

Comment:
More accurately, it's if there are multiple possible median elements that the indices returned are nondeterminism. This is probably something we should fix, by the way. We fixed argmin and argmax, for example, to always return the indices of the first minimum or maximum element.

Reply:
I'm happy to see the docs actually explain this: 

indices does not necessarily contain the first occurrence of each median value found, unless it is unique. The exact implementation details are device-specific. Do not expect the same result when run on CPU and GPU in general. For the same reason do not expect the gradients to be deterministic.